README.md

```diff
@@ -31,9 +41,277 @@ docker run \
+    "COPY items (embedding) FROM STDIN (FORMAT BINARY)"
+) as copy:
+    copy.write_row([np.array([1, 2, 3])])
+```
+
+Add an approximate index
+```python
+from pgvecto_rs.types import IndexOption, Hnsw, Ivf
+
+conn.execute(
+    "CREATE INDEX emb_idx_1 ON items USING \
+        vectors (embedding vector_l2_ops) WITH (options=$${}$$);".format(
+        IndexOption(index=Hnsw(), threads=1).dumps()
+    ),
+)
+# or
+conn.execute(
+    "CREATE INDEX emb_idx_2 ON items USING \
+        vectors (embedding vector_l2_ops) WITH (options=$${}$$);".format(
+        IndexOption(index=Ivf()).dumps()
+    ),
+)
+# Apply all changes
+conn.commit()
+```
+
+Get the nearest neighbors to a vector
+```python
+conn.execute('SELECT * FROM items ORDER BY embedding <-> %s LIMIT 5', (embedding,)).fetchall()
+```
+
+Get the distance
+```python
+conn.execute('SELECT embedding <-> %s FROM items \
+    ORDER BY embedding <-> %s', (embedding, embedding)).fetchall()
+```
+
+Get items within a certain distance
+```python
+conn.execute('SELECT * FROM items WHERE embedding <-> %s < 1.0 \
+    ORDER BY embedding <-> %s', (embedding, embedding)).fetchall()
+```
+
+See [examples/psycopg_example.py](examples/psycopg_example.py) and [tests/test_psycopg.py](tests/test_psycopg.py) for more examples
+
+### Django
+
+Install dependencies:
+
+```bash
+pip install "pgvecto_rs[django]"
+```
+
+Create a migration to enable the extension
+
+```python
+from pgvecto_rs.django import VectorExtension
+
+class Migration(migrations.Migration):
+    operations = [
+        VectorExtension()
+    ]
+```
+
+Add a vector field to your model
+
+```python
+from pgvecto_rs.django import VectorField
+
+class Document(models.Model):
+    embedding = VectorField(dimensions=3)
 ```
 
+All supported types are shown in this table
+
+| Native types | Types for Django   | Correspond to pgvector-python |
+| ------------ | ------------------ | ----------------------------- |
+| vector       | VectorField        | VectorField                   |
+| svector      | SparseVectorField  | SparseVectorField             |
+| vecf16       | Float16VectorField | HalfVectorField               |
+| bvector      | BinaryVectorField  | BitField                      |
+
+Insert a vector
+```python
+Item(embedding=[1, 2, 3]).save()
+```
+
+Add an approximate index
+```python
+from django.db import models
+from pgvecto_rs.django import HnswIndex, IvfIndex
+from pgvecto_rs.types import IndexOption, Hnsw
+
+
+class Item(models.Model):
+    class Meta:
+        indexes = [
+            HnswIndex(
+                name="emb_idx_1",
+                fields=["embedding"],
+                opclasses=["vector_l2_ops"],
+                # don't pass any of `m`, `ef_construction`, `threads`, `quantization_type` or `quantization_ratio`
+                # if created by `with_option`, they will be overridden
+            ).with_option(
+                IndexOption(index=Hnsw(m=16, ef_construction=100), threads=1)
+            ),
```

Comment:
When do we need to call the `with_option`? I don't know if there is such kind of scenario. Only the `threads` can be changed on the fly.

Reply:
I think people may want to create Index by a structured `IndexOption` instead of a set of arguments before, but as you say, it introduces more complex. I would remove this usage.